### PR TITLE
Migrate to clap

### LINKED ISSUE #15: Migrate from `structopt` to `clap`
`clap` has recently integrated `structopt` directly via the `derive` feature flag. As `structopt` is now in [maintenance mode](https://github.com/TeXitoi/structopt#maintenance) it should be preferable to make the switch.

### PR DESCRIPTION
**Pull request by nezuo**
Fixes #15. 

Note that clap has colors enabled unlike structopt.

### CODE REVIEW

File: src/cli/mod.rs [hunk 1]
```diff
@@ -13,9 +13,10 @@ use crate::tool_spec::ToolSpec;
 use crate::tool_storage::{InstalledToolsCache, ToolStorage};
 use crate::trust::{TrustCache, TrustMode};
 
-#[derive(Debug, StructOpt)]
+#[derive(Debug, Parser)]
+#[clap(version)]
```
Comment (LPGhatguy) [inline src/cli/mod.rs]:
```suggestion
#[clap(author, version, about)]
```

I don't know if clap does these things by default like structopt does?

**Review by LPGhatguy — COMMENTED** [review]
One question and then LGTM!

File: src/cli/mod.rs (hunk 1 above)
Comment (nezuo) [reply src/cli/mod.rs]: These aren't on by default but they didn't appear to be on before. Should I add an you as an author in the `Cargo.toml` as well then?

Comment (LPGhatguy) [reply src/cli/mod.rs]: Yeah, good call. Thank you!

**Comment by LPGhatguy** [comment]
Thank you!

**Pull request merged**